nbs/15_gsa_gls/20-null_simulations/10_gwas_harmonization/README.md

```diff
@@ -135,17 +151,112 @@ mkdir -p _tmp/postprocessing
 cat cluster_jobs/10_postprocessing_job.sh | bsub
 ```
 
-Check logs with: `bash check_job.sh -i _tmp/postprocessing`
+Check logs with:
+```bash
+bash check_job.sh -i _tmp/postprocessing
+```
 
-There should be 100 files in the output directory, one for each random phenotype.
+Another check is to count how many parts were processed for each random phenotype.
+It should be 22 chromosomes times 10 batches (220):
+```bash
+bash check_job.sh \
+  -i _tmp/postprocessing/ \
+  -p "INFO - Processing imputed random" \
+  -c 220
+
+# which should output:
+# Finished checking [NUMBER_OF_PHENOTYPES] logs:
+#  All jobs finished successfully
+```
+
+
+## Compute set of common variants
+
+For some reason, the postprocessing scripts previously run do not generate exactly the same set of final variants for imputed GWAS.
+Each final GWAS should have the same set of imputed variants, since they were all imputed using GWAS from the same cohort and exactly the same input variants.
+However, this is not the case, and for this null simulations we need all GWAS to have the same set of variants to efficiently compute gene correlations.
+
+So what we do is to read all final GWAS files, generate a set of common variants across all of them, and then align all final GWAS to that.
+
+
+```python
+import pickle
+from pathlib import Path
+import concurrent
+
+import numpy as np
+import pandas as pd
+
+import conf
+
+N_SAMPLES = 50
+
+POST_IMPUTED_DIR = Path(
+  conf.RESULTS["GLS_NULL_SIMS"],
+  "post_imputed_gwas"
+).resolve()
+assert POST_IMPUTED_DIR.exists(), POST_IMPUTED_DIR
+
+input_files = sorted(list(POST_IMPUTED_DIR.glob("*.txt.gz")))
+len(input_files)
+
+# sample files
+np.random.seed(0)
+input_files = np.random.choice(input_files, size=N_SAMPLES, replace=False)
+len(input_files)
+
+# read all GWAS and find a set of common panel_variant_id_values
+def _get_gwas_variants(f):
```

Comment:
You have much more software engineering experience than I do so I'm curious --  I am not used to seeing functions embedded in the main code. When/why do you do that?

Reply:
Good question. It's not common to have a function in a bunch of code intended to be copied/pasted into an ipython console (unless it's helpful for readability). But in this case, I'm using parallelization with the `concurrent.futures` package, which needs a function that works on a single chunk of the entire data. That's why I've added this function, which is used when calling `executor.map`.